File: python/count_binary_substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        binarySubstrings, previous, current = 0, 0, 1
        index = 1
        while index < len(s):
            if s[index - 1] != s[index]:
                binarySubstrings += min(previous, current)
                previous, current = current, 1
            else: current += 1
            index += 1
        return binarySubstrings + min(previous, current)

File: src/CountBinarySubstrings.java
public class CountBinarySubstrings {
    public int countBinarySubstrings(String string) {
        int binarySubStrings = 0, prev = 0, current = 1;
        for (int index = 1 ; index < string.length() ; index++) {
            if (string.charAt(index - 1) != string.charAt(index)) {
                binarySubStrings += Math.min(prev, current);
                prev = current;
                current = 1;
            } else current++;
        }
        return binarySubStrings + Math.min(prev, current);
    }
}
